### ISSUE TITLE
WS - Members - Inviting new user is duplicated in members list

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#22747]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/22075
## Action Performed:
Pre: Create new accounts

1. New task assignee with rybkina+09@gmail.com and rybkina@gmail.com
2. Invite users to workspaces with rybkina+09@gmail.com and rybkina@gmail.com
3. New 1:1 chat create with rybkina+09@gmail.com and rybkina@gmail.com
4. New group chat create with rybkina+09@gmail.com and rybkina@gmail.com

## Expected Result:
Avatars & tooltips appear correctly (these accounts should all show defaults)
Display names / logins show where they should (not showing as blank anywhere)

## Actual Result:
1. Avatar doesn't show on assign page
2. User shown 2 times
3. Sometimes title and avatar missing

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.39.5

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/6732b23f-5d69-4bae-8100-67403fb127cc

<img width="300" src="https://github.com/Expensify/App/assets/78819774/acbb2db0-e0e7-4df3-b6f0-fc9fea5314a3" />

https://github.com/Expensify/App/assets/78819774/a56dd25a-92ca-44ae-8e29-c6f9b9f577a1

<img width="300" src="https://github.com/Expensify/App/assets/78819774/fe875b85-ef22-4fd6-bfc8-1d7d445a6dd9" />

<img width="300" src="https://github.com/Expensify/App/assets/78819774/a8f0c760-19fe-4270-b236-8a0712faebde" />

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by arosiclair** [comment]
Attempting to repro though these steps are not written well and there seems to be two maybe 3 issues here

**Comment by arosiclair** [comment]
Okay I was able to reproduce the assignee's avatar and display name not updating in the Confirm task on **v1.3.39-10**:

![Screenshot 2023-07-12 at 2 13 29 PM](https://github.com/Expensify/App/assets/6833644/933c198b-cdf8-445b-a81f-2f5b006b3cd2)

I also reproduced the duplicate workspace members displaying after inviting them:

![Screenshot 2023-07-12 at 2 17 10 PM](https://github.com/Expensify/App/assets/6833644/0bc8048f-ba6f-4dda-8bcb-8beae76bcac1)

**Comment by arosiclair** [comment]
I reproduced both of the above issues in prod **v1.3.38-7**. Removing the deploy blocker label

> 3. Sometimes title and avatar missing

I was not able to reproduce this on staging or prod.

**Comment by arosiclair** [comment]
> 1. Avatar doesn't show on assign page

This seems to be the same issue as #22293 

> 2. User shown 2 times [in workspace members list after inviting]

I'm not finding any existing issues for this. Let's keep this issue to address that bug specifically

**Comment by arosiclair** [comment]
TL;DR: 

1. Log into NewDot
2. Create a workspace
3. Invite 2 new users you haven't chatted with before (on a different domain)
4. Observe the members get duplicated in the list:

![Screenshot 2023-07-12 at 2 56 58 PM](https://github.com/Expensify/App/assets/6833644/0b618bb6-9237-4ddd-9a06-9d5f762dfd07)

**Comment by trjExpensify** [comment]
Cool, so what are we doing here? Continuing on #21706?

**Comment by trjExpensify** [comment]
Alrighty, closing!
